src/compiler/sst.rs

```diff
@@ -0,0 +1,132 @@
+use crate::common::{
+    span::Spanned,
+    data::Data,
+};
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct UniqueSymbol(pub usize);
+
+/// A pattern that mirrors the structure of some Data.
+#[derive(Debug, Clone, PartialEq)]
+pub enum SSTPattern {
+    Symbol(UniqueSymbol),
+    Data(Data),
+    Label(String, Box<Spanned<SSTPattern>>), // todo usize for label
+    Tuple(Vec<Spanned<SSTPattern>>),
+    // Where {
```

Comment:
Debug comments?

Reply:
This is a planned feature; just have it commented out for now.